### ISSUE TITLE
Blink strenght in neurosky mindwave

### ISSUE THREAD

**Issue by sagniknitr** [lihas/NeuroPy#8]
Hi I am using neurosky mindwave module and connected it to arduino using bluetooth module.I am getting the attention and meditation value but the blink state in showing 0.

Please help in identifying the problem.

**Comment by lihas** [comment]
Just to be sure, you are blinking as well, when you are monitoring the value?
Also, did you try neurosky's tools, do they show blink value?

**Comment by thebird76** [comment]
Hi I am also having issues getting the value from blinking.  Attention and meditation are working for me but blinkStrength never seems to go above zero.

**Comment by lihas** [comment]
I'll check it next weekend, i.e. the coming one. Meanwhile can you confirm if you are able to see blinking in the tools provided by Neurosky?

**Comment by StenXe** [comment]
Even I am experiencing same issue using NeuroPy. Our final year project depends on the success of this out of the world module.  A

**Comment by StenXe** [comment]
Even I experienced the same issue. I did a bit of digging and found out that 32 size payload doesn't contain code '16' which is for blink strength, it contains is '05' which is for meditation and '00' at last, after that I suspect would be blink strength value. Maybe the payload size falls short to actually record the blink strength value.
EDIT: After a lot of research I have understood NeuroPy is incapable of obtaining blink strength because it communicates on bluetooth serial port which doesn't output blink value. For obtaining blink  strength one might need to use Thinkgear Connector which processes the blinkStrength and passes on to client connected via socket. For more details check Documentation.
http://developer.neurosky.com/docs/doku.php?id=thinkgear_communications_protocol
http://developer.neurosky.com/docs/doku.php?id=thinkgear_connector_tgc

Correct me if I am wrong.

**Comment by kushal12345** [comment]
So is this problem solved? help needed if yes

**Comment by StenXe** [comment]
No, as I said NeuroPy by itself won't give you the blinkStrength value. You'll have to use ThinkGear Connector to read blink values from the device using socket.

**Comment by kushal12345** [comment]
StenXe  can you help me  with the details of using  thinkgear connector in raspberry pi

**Comment by StenXe** [comment]
I didn't work on Raspbery Pi, I used laptop with Windows 10 on it. As far as I know, ThinkGearConnector is officially supported for Windows and MacOS but there is 3rd Party support for Linux, which I am not sure of.
If you want to use it on Raspberry Pi either try to work around with 3rd party developers (as mentioned in the second link of my previous post) or use a device that can run full Windows/Mac OS and run the ThinkGearConnector on it.
